src/main/java/com/zendesk/maxwell/schema/MysqlSavedSchema.java

```diff
@@ -226,19 +226,30 @@ private void executeColumnInsert(Connection conn, ArrayList<Object> columnData)
 		columnData.clear();
 	}
 
-	public static MysqlSavedSchema restore(Connection connection, MaxwellContext context) throws SQLException, IOException, InvalidSchemaError {
-		MysqlSavedSchema s = new MysqlSavedSchema(context.getServerID(), context.getCaseSensitivity());
+	public static MysqlSavedSchema restore(MaxwellContext context, BinlogPosition targetPosition) throws SQLException, InvalidSchemaError {
+		try ( Connection conn = context.getMaxwellConnection() ) {
+			Long schemaID = findSchema(conn, targetPosition, context.getServerID());
+			if (schemaID == null)
+				return null;
 
-		s.restoreFrom(connection, context.getInitialPosition());
+			MysqlSavedSchema savedSchema = new MysqlSavedSchema(context.getServerID(), context.getCaseSensitivity());
 
-		return s;
+			savedSchema.restoreFromSchemaID(conn, schemaID);
+			savedSchema.handleVersionUpgrades(conn);
+
+			return savedSchema;
+		}
 	}
 
-	private List<ResolvedSchemaChange> parseDeltas(String json) throws IOException {
+	private List<ResolvedSchemaChange> parseDeltas(String json) {
 		if ( json == null )
 			return null;
 
-		return mapper.readerFor(listOfResolvedSchemaChangeType).readValue(json.getBytes());
+		try {
+			return mapper.readerFor(listOfResolvedSchemaChangeType).readValue(json.getBytes());
+		} catch ( IOException e ) {
+			throw new RuntimeException("couldn't parse json delta: " + json.getBytes(), e);
+		}
```

Comment:
Stopped this IOException from leaking up the stack, as it's not gonna be thrown 99.999% of the time.  Best to just crash.